src/SamplesApp/SamplesApp.UITests/TestFramework/ImageAssert.cs

```diff
@@ -15,32 +15,29 @@ namespace SamplesApp.UITests.TestFramework
 {
 	public static class ImageAssert
 	{
-		public static void AreEqual(FileInfo expected, FileInfo actual, IAppRect rect)
-			=> AreEqual(expected, actual, new Rectangle((int)rect.X, (int)rect.Y, (int)rect.Width, (int)rect.Height));
+		private static readonly Rectangle FirstQuadrant = new Rectangle(0, 0, int.MaxValue, int.MaxValue);
 
-		public static void AreEqual(FileInfo expected, FileInfo actual, Rectangle? rect = null)
-		{
-			rect = rect ?? new Rectangle(0, 0, int.MaxValue, int.MaxValue);
-			AreEqual(expected, rect.Value, actual, rect.Value);
-		}
+		#region AreEqual
+		public static void AreEqual(FileInfo expected, FileInfo actual, IAppRect rect, [CallerLineNumber] int line = 0)
+			=> AreEqualImpl(expected, rect.ToRectangle(), actual, rect.ToRectangle(), line);
+
+		public static void AreEqual(FileInfo expected, FileInfo actual, Rectangle? rect = null, [CallerLineNumber] int line = 0)
+			=> AreEqualImpl(expected, rect ?? FirstQuadrant, actual, rect ?? FirstQuadrant, line);
 
-		public static void AreEqual(FileInfo expected, IAppRect expectedRect, FileInfo actual, IAppRect actualRect)
-			=> AreEqual(
-				expected,
-				new Rectangle((int)expectedRect.X, (int)expectedRect.Y, (int)expectedRect.Width, (int)expectedRect.Height),
-				actual,
-				new Rectangle((int)actualRect.X, (int)actualRect.Y, (int)actualRect.Width, (int)actualRect.Height));
+		public static void AreEqual(FileInfo expected, IAppRect expectedRect, FileInfo actual, IAppRect actualRect, [CallerLineNumber] int line = 0)
+			=> AreEqualImpl(expected, expectedRect.ToRectangle(), actual, actualRect.ToRectangle(), line);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'actualRect' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5170992)